### PR TITLE
[Table sanity check] Integration test for the `known_hosts` table

### PR DESCRIPTION
**Pull request by akindyakov**
Fixes: #4879

### CODE REVIEW

File: osquery/tables/system/posix/tests/known_hosts_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  namespace fs = boost::filesystem;
+
+  auto results = QueryData{};
+  auto directory = fs::temp_directory_path() /
+                   fs::unique_path("osquery.known_hosts_impl_tests.%%%%-%%%%");
+
+  ASSERT_TRUE(fs::create_directories(directory));
+
+  auto const path_guard =
+      scope_guard::create([directory]() { fs::remove_all(directory); });
```
Comment (guliashvili) [inline osquery/tables/system/posix/tests/known_hosts_tests.cpp]: SetUp 
TearDown
Can be used for that.

File: osquery/tests/integration/tables/known_hosts.cpp [hunk 2]
```diff
@@ -16,25 +16,16 @@
...
-  // 4. Perform validation
-  // validate_rows(data, row_map);
+TEST_F(KnownHostsTest, sanity) {
+  QueryData rows = execute_query("select * from known_hosts");
+  ValidatatioMap row_map = {
+      {"uid", IntType},
+      {"key", NonEmptyString},
+      {"key_file", FileOnDisk},
+  };
+  validate_rows(rows, row_map);
```
Comment (fmanco) [inline osquery/tests/integration/tables/known_hosts.cpp]: Check if rows is empty and if so LOG(WARNING) that table won't be actually tested.

Comment (fmanco) [reply osquery/tests/integration/tables/known_hosts.cpp]: Maybe we could even put that under `validate_rows()`.

File: osquery/tables/system/posix/tests/known_hosts_tests.cpp (hunk 1 above)
Comment (akindyakov) [reply osquery/tables/system/posix/tests/known_hosts_tests.cpp]: Yes, I know. But `SetUp/TearDown` do it for all tests in the file, but I need it only for one. Of course we have only one test and it does not make a lot of sense, but eventually we will add more and it make a big mess in `SetUp/TearDown`.
Therefore, if you don't have a strict objections I'd like to leave it like this. What do you think?

**Pull request merged**